### PR TITLE
GDPRConsent added to Register

### PR DESCRIPTION
**Pull request by swetlasg**

### CODE REVIEW

File: src/state_types/models/context.rs [hunk 1]
```diff
@@ -105,9 +106,15 @@ impl<Env: Environment + 'static> Update for Ctx<Env> {
                         None => Effects::msg(CtxUpdate(new_content).into()).unchanged(),
                     }
                 }
-                ActionUser::Register { email, password } => Effects::one(authenticate::<Env>(
+                ActionUser::Register { email, password, tos, privacy, marketing, from } => Effects::one(authenticate::<Env>(
                     action.to_owned(),
-                    APIRequest::Register { email, password },
+                    APIRequest::Register { email, password, gdpr_consent: GDPRConsent {
```
Comment (Ivshti) [inline src/state_types/models/context.rs]: you can do tihs like JS: just specify `{ tos, privacy, ` etc.

File: src/state_types/msg/actions.rs [hunk 2]
```diff
@@ -37,7 +37,7 @@ pub enum ActionSettings {
 #[serde(tag = "userOp", content = "args")]
 pub enum ActionUser {
     Login { email: String, password: String },
-    Register { email: String, password: String },
+    Register { email: String, password: String, tos: bool, privacy: bool, marketing: bool, from: String },
```
Comment (Ivshti) [inline src/state_types/msg/actions.rs]: why is this not in a `gdpr_consent` field?

File: src/state_types/models/context.rs (hunk 1 above)
Comment (Ivshti) [reply src/state_types/models/context.rs]: if you implement my second comment, then this won't be needed

File: src/state_types/msg/actions.rs (hunk 2 above)
Comment (NikolaBorislavovHristov) [reply src/state_types/msg/actions.rs]: because we create the date in the core

Comment (NikolaBorislavovHristov) [reply src/state_types/msg/actions.rs]: and the ui dont care about how data is structured

Comment (Ivshti) [reply src/state_types/msg/actions.rs]: this is overkill; create the date in the UI

**Pull request merged**